### DISCUSSION TITLE
install_help

### DISCUSSION THREAD

**Discussion by joejiang2023** [danielver02/ALPS#152] [Q&A] [answered]
Hi all,

I'm trying to install the newly released ALPS version here and failed to install it properly. Could anyone help to provide any clues? There could be some library dependency not being installed according to the configure prompt. Not sure what is the key issue here.
Here is what my Mac (Catalina 10.15.7) complains about when I try to make install it:

**(base)MacBook-Pro ALPS-master % ./configure**
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... ./install-sh -c -d
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking whether gcc understands -c and -o together... yes
checking for style of include used by make... GNU
checking dependency style of gcc... none
checking for mpifort... mpifort
checking whether we are using the GNU Fortran compiler... yes
checking whether mpifort accepts -g... yes
**checking for library containing dgesv... no
checking for library containing dgesv... (cached) no
checking for library containing dgesv... (cached) no**
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating src/Makefile
config.status: creating interpolation/Makefile
config.status: creating distribution/Makefile
config.status: executing depfiles commands

**(base) MacBook-Pro ALPS-master % make**
Making all in src
mpifort -O4 -fbounds-check -ffpe-trap=zero,overflow -Wunused -funroll-loops -g -fbacktrace  -c -o ../distribution/distribution_analyt.o ../distribution/distribution_analyt.f90
mpifort -O4 -fbounds-check -ffpe-trap=zero,overflow -Wunused -funroll-loops -g -fbacktrace  -c -o ALPS_var.o ALPS_var.f90
mpifort -O4 -fbounds-check -ffpe-trap=zero,overflow -Wunused -funroll-loops -g -fbacktrace  -c -o ALPS_io.o ALPS_io.f90
mpifort -O4 -fbounds-check -ffpe-trap=zero,overflow -Wunused -funroll-loops -g -fbacktrace  -c -o ALPS_com.o ALPS_com.f90
mpifort -O4 -fbounds-check -ffpe-trap=zero,overflow -Wunused -funroll-loops -g -fbacktrace  -c -o ALPS_analyt.o ALPS_analyt.f90
mpifort -O4 -fbounds-check -ffpe-trap=zero,overflow -Wunused -funroll-loops -g -fbacktrace  -c -o ALPS_NHDS.o ALPS_NHDS.f90
mpifort -O4 -fbounds-check -ffpe-trap=zero,overflow -Wunused -funroll-loops -g -fbacktrace  -c -o ALPS_fns_rel.o ALPS_fns_rel.f90
mpifort -O4 -fbounds-check -ffpe-trap=zero,overflow -Wunused -funroll-loops -g -fbacktrace  -c -o ALPS_fns.o ALPS_fns.f90
mpifort -O4 -fbounds-check -ffpe-trap=zero,overflow -Wunused -funroll-loops -g -fbacktrace  -c -o ALPS.o ALPS.f90
mpifort -O4 -fbounds-check -ffpe-trap=zero,overflow -Wunused -funroll-loops -g -fbacktrace    -o ALPS ../distribution/distribution_analyt.o ALPS_var.o ALPS_io.o ALPS_com.o ALPS_analyt.o ALPS_NHDS.o ALPS_fns_rel.o ALPS_fns.o ALPS.o -lm -llapack -lblas 
Making all in distribution
mpifort -O4 -fcheck=all -fbacktrace -fbounds-check -ffpe-trap=zero,overflow -Wconversion -Wsurprising -ffixed-line-length-none -ffree-line-length-none -Wunused -g -fbacktrace -I../src  -c -o generate_distribution.o generate_distribution.f90
mpifort -O4 -fcheck=all -fbacktrace -fbounds-check -ffpe-trap=zero,overflow -Wconversion -Wsurprising -ffixed-line-length-none -ffree-line-length-none -Wunused -g -fbacktrace -I../src    -o generate_distribution distribution_analyt.o generate_distribution.o -lm -llapack -lblas 
Making all in interpolation
mpifort -O4 -fcheck=all -fbacktrace -fbounds-check -ffpe-trap=zero,overflow -Wconversion -Wsurprising -ffixed-line-length-none -ffree-line-length-none -Wunused -g -fbacktrace  -c -o interpolation.o interpolation.f90
mpifort -O4 -fcheck=all -fbacktrace -fbounds-check -ffpe-trap=zero,overflow -Wconversion -Wsurprising -ffixed-line-length-none -ffree-line-length-none -Wunused -g -fbacktrace    -o interpolation interpolation.o -lm -llapack -lblas 
make[1]: Nothing to be done for `all-am'.

**(base) MacBook-Pro ALPS-master % make install**
Making install in src
 .././install-sh -c -d '/usr/local/bin'
  /usr/bin/install -c ALPS '/usr/local/bin'
make[2]: Nothing to be done for `install-data-am'.
Making install in distribution
 .././install-sh -c -d '/usr/local/bin'
  /usr/bin/install -c generate_distribution '/usr/local/bin'
make[2]: Nothing to be done for `install-data-am'.
Making install in interpolation
 .././install-sh -c -d '/usr/local/bin'
  /usr/bin/install -c interpolation '/usr/local/bin'
make[2]: Nothing to be done for `install-data-am'.
make[2]: Nothing to be done for `install-exec-am'.
make[2]: Nothing to be done for `install-data-am'.

**Comment by danielver02** [comment]
Hi Joe,

See if you have the LAPACK and BLAS libraries installed. That’s where the dgesv error is coming from. Apart from that, the compilation looks fine. 

Best wishes,
—Daniel

**Reply by joejiang2023** [reply]
Hi Daniel,

I've installed LAPACK and BLAS, but the ./configure still can't find their libraries. I tried to google how to link the libraries to it, but it's still reporting the dgesv error. Could you please help to look at it once more?

Thanks,
-Joe

Here is log info:
PATH: /Users/joejiang/usr/local/Cellar/lapack/3.11/lib

## ----------- ##
## Core tests. ##
## ----------- ##

configure:1923: checking for a BSD-compatible install
configure:1991: result: /usr/bin/install -c
configure:2002: checking whether build environment is sane
configure:2057: result: yes
configure:2208: checking for a thread-safe mkdir -p
configure:2247: result: ./install-sh -c -d
configure:2254: checking for gawk
configure:2270: found /usr/local/bin/gawk
configure:2281: result: gawk
configure:2292: checking whether make sets $(MAKE)
configure:2314: result: yes
configure:2343: checking whether make supports nested variables
configure:2360: result: yes
configure:2547: checking for gcc
configure:2563: found /usr/bin/gcc
configure:2574: result: gcc
configure:2605: checking for C compiler version
configure:2614: gcc --version >&5
Apple clang version 12.0.0 (clang-1200.0.32.29)
Target: x86_64-apple-darwin19.6.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
Configured with: --prefix=/Library/Developer/CommandLineTools/usr --with-gxx-include-dir=/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/c++/4.2.1
configure:2625: $? = 0
configure:2614: gcc -v >&5
Configured with: --prefix=/Library/Developer/CommandLineTools/usr --with-gxx-include-dir=/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/c++/4.2.1
Apple clang version 12.0.0 (clang-1200.0.32.29)
Target: x86_64-apple-darwin19.6.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
configure:2625: $? = 0
configure:2614: gcc -V >&5
clang: error: argument to '-V' is missing (expected 1 value)
clang: error: no input files
configure:2625: $? = 1
configure:2614: gcc -qversion >&5
clang: error: unknown argument '-qversion'; did you mean '--version'?
clang: error: no input files
configure:2625: $? = 1
configure:2645: checking whether the C compiler works
configure:2667: gcc    conftest.c  >&5
configure:2671: $? = 0
configure:2719: result: yes
configure:2722: checking for C compiler default output file name
configure:2724: result: a.out
configure:2730: checking for suffix of executables
configure:2737: gcc -o conftest    conftest.c  >&5
configure:2741: $? = 0
configure:2763: result: 
configure:2785: checking whether we are cross compiling
configure:2793: gcc -o conftest    conftest.c  >&5
configure:2797: $? = 0
configure:2804: ./conftest
configure:2808: $? = 0
configure:2823: result: no
configure:2828: checking for suffix of object files
configure:2850: gcc -c   conftest.c >&5
configure:2854: $? = 0
configure:2875: result: o
configure:2879: checking whether we are using the GNU C compiler
configure:2898: gcc -c   conftest.c >&5
configure:2898: $? = 0
configure:2907: result: yes
configure:2916: checking whether gcc accepts -g
configure:2936: gcc -c -g  conftest.c >&5
configure:2936: $? = 0
configure:2977: result: yes
configure:2994: checking for gcc option to accept ISO C89
configure:3057: gcc  -c -g -O2  conftest.c >&5
configure:3057: $? = 0
configure:3070: result: none needed
configure:3095: checking whether gcc understands -c and -o together
configure:3117: gcc -c conftest.c -o conftest2.o
configure:3120: $? = 0
configure:3117: gcc -c conftest.c -o conftest2.o
configure:3120: $? = 0
configure:3132: result: yes
configure:3160: checking for style of include used by make
configure:3188: result: GNU
configure:3214: checking dependency style of gcc
configure:3325: result: none
configure:3393: checking for mpifort
configure:3409: found /usr/local/bin/mpifort
configure:3420: result: mpifort
configure:3446: checking for Fortran compiler version
configure:3455: mpifort --version >&5
GNU Fortran (Homebrew GCC 13.1.0) 13.1.0
Copyright (C) 2023 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

configure:3466: $? = 0
configure:3455: mpifort -v >&5
Using built-in specs.
COLLECT_GCC=/usr/local/bin/gfortran
COLLECT_LTO_WRAPPER=/usr/local/Cellar/gcc/13.1.0/bin/../libexec/gcc/x86_64-apple-darwin19/13/lto-wrapper
Target: x86_64-apple-darwin19
Configured with: ../configure --prefix=/usr/local/opt/gcc --libdir=/usr/local/opt/gcc/lib/gcc/current --disable-nls --enable-checking=release --with-gcc-major-version-only --enable-languages=c,c++,objc,obj-c++,fortran --program-suffix=-13 --with-gmp=/usr/local/opt/gmp --with-mpfr=/usr/local/opt/mpfr --with-mpc=/usr/local/opt/libmpc --with-isl=/usr/local/opt/isl --with-zstd=/usr/local/opt/zstd --with-pkgversion='Homebrew GCC 13.1.0' --with-bugurl=https://github.com/Homebrew/homebrew-core/issues --with-system-zlib --build=x86_64-apple-darwin19 --with-sysroot=/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk
Thread model: posix
Supported LTO compression algorithms: zlib zstd
gcc version 13.1.0 (Homebrew GCC 13.1.0) 
configure:3466: $? = 0
configure:3455: mpifort -V >&5
gfortran: error: unrecognized command-line option '-V'
gfortran: fatal error: no input files
compilation terminated.
configure:3466: $? = 1
configure:3455: mpifort -qversion >&5
gfortran: error: unrecognized command-line option '-qversion'; did you mean '--version'?
gfortran: fatal error: no input files
compilation terminated.
configure:3466: $? = 1
configure:3475: checking whether we are using the GNU Fortran compiler
configure:3488: mpifort -c   conftest.F >&5
configure:3488: $? = 0
configure:3497: result: yes
configure:3503: checking whether mpifort accepts -g
configure:3514: mpifort -c -g  conftest.f >&5
configure:3514: $? = 0
configure:3522: result: yes
configure:3561: checking for library containing dgesv
configure:3579: mpifort -o conftest    conftest.f  >&5
Undefined symbols for architecture x86_64:
  "_dgesv_", referenced from:
      _MAIN__ in cc2bAMST.o
ld: symbol(s) not found for architecture x86_64
collect2: error: ld returned 1 exit status
configure:3579: $? = 1
configure: failed program was:
|       program main
|       call dgesv
|       end
configure:3579: mpifort -o conftest    conftest.f -lblas   >&5
Undefined symbols for architecture x86_64:
  "_dgesv_", referenced from:
      _MAIN__ in ccwZTIZm.o
ld: symbol(s) not found for architecture x86_64
collect2: error: ld returned 1 exit status
configure:3579: $? = 1
configure: failed program was:
|       program main
|       call dgesv
|       end
configure:3596: result: no
configure:3604: checking for library containing dgesv
configure:3639: result: no
configure:3647: checking for library containing dgesv
configure:3682: result: no
configure:3849: checking that generated files are newer than configure
configure:3855: result: done
configure:3878: creating ./config.status

## ---------------------- ##
## Running config.status. ##
## ---------------------- ##

This file was extended by ALPS config.status 1.0, which was
generated by GNU Autoconf 2.69.  Invocation command line was

  CONFIG_FILES    = 
  CONFIG_HEADERS  = 
  CONFIG_LINKS    = 
  CONFIG_COMMANDS = 
  $ ./config.status 

on joejiangdeMacBook-Pro.local

config.status:778: creating Makefile
config.status:778: creating src/Makefile
config.status:778: creating interpolation/Makefile
config.status:778: creating distribution/Makefile
config.status:950: executing depfiles commands

## ---------------- ##
## Cache variables. ##
## ---------------- ##

ac_cv_c_compiler_gnu=yes
ac_cv_env_CC_set=
ac_cv_env_CC_value=
ac_cv_env_CFLAGS_set=
ac_cv_env_CFLAGS_value=
ac_cv_env_CPPFLAGS_set=
ac_cv_env_CPPFLAGS_value=
ac_cv_env_FCFLAGS_set=
ac_cv_env_FCFLAGS_value=
ac_cv_env_FC_set=
ac_cv_env_FC_value=
ac_cv_env_LDFLAGS_set=
ac_cv_env_LDFLAGS_value=
ac_cv_env_LIBS_set=
ac_cv_env_LIBS_value=
ac_cv_env_build_alias_set=
ac_cv_env_build_alias_value=
ac_cv_env_host_alias_set=
ac_cv_env_host_alias_value=
ac_cv_env_target_alias_set=
ac_cv_env_target_alias_value=
ac_cv_fc_compiler_gnu=yes
ac_cv_objext=o
ac_cv_path_install='/usr/bin/install -c'
ac_cv_prog_AWK=gawk
ac_cv_prog_ac_ct_CC=gcc
ac_cv_prog_ac_ct_FC=mpifort
ac_cv_prog_cc_c89=
ac_cv_prog_cc_g=yes
ac_cv_prog_fc_g=yes
ac_cv_prog_make_make_set=yes
ac_cv_search_dgesv=no
am_cv_CC_dependencies_compiler_type=none
am_cv_make_support_nested_variables=yes
am_cv_prog_cc_c_o=yes

## ----------------- ##
## Output variables. ##
## ----------------- ##

ACLOCAL='${SHELL} /Users/joejiang/documents/ALPS/ALPS-master/missing aclocal-1.15'
AMDEPBACKSLASH='\'
AMDEP_FALSE='#'
AMDEP_TRUE=''
AMTAR='$${TAR-tar}'
AM_BACKSLASH='\'
AM_DEFAULT_V='$(AM_DEFAULT_VERBOSITY)'
AM_DEFAULT_VERBOSITY='1'
AM_V='$(V)'
AUTOCONF='${SHELL} /Users/joejiang/documents/ALPS/ALPS-master/missing autoconf'
AUTOHEADER='${SHELL} /Users/joejiang/documents/ALPS/ALPS-master/missing autoheader'
AUTOMAKE='${SHELL} /Users/joejiang/documents/ALPS/ALPS-master/missing automake-1.15'
AWK='gawk'
CC='gcc'
CCDEPMODE='depmode=none'
CFLAGS='-g -O2'
CPPFLAGS=''
CYGPATH_W='echo'
DEFS='-DPACKAGE_NAME=\"ALPS\" -DPACKAGE_TARNAME=\"alps\" -DPACKAGE_VERSION=\"1.0\" -DPACKAGE_STRING=\"ALPS\ 1.0\" -DPACKAGE_BUGREPORT=\"d.verscharen@ucl.ac.uk\" -DPACKAGE_URL=\"\" -DPACKAGE=\"alps\" -DVERSION=\"1.0\"'
DEPDIR='.deps'
ECHO_C='\c'
ECHO_N=''
ECHO_T=''
EXEEXT=''
FC='mpifort'
FCFLAGS=''
INSTALL_DATA='${INSTALL} -m 644'
INSTALL_PROGRAM='${INSTALL}'
INSTALL_SCRIPT='${INSTALL}'
INSTALL_STRIP_PROGRAM='$(install_sh) -c -s'
LDFLAGS=''
LIBOBJS=''
LIBS=''
LTLIBOBJS=''
MAKEINFO='${SHELL} /Users/joejiang/documents/ALPS/ALPS-master/missing makeinfo'
MKDIR_P='./install-sh -c -d'
OBJEXT='o'
PACKAGE='alps'
PACKAGE_BUGREPORT='d.verscharen@ucl.ac.uk'
PACKAGE_NAME='ALPS'
PACKAGE_STRING='ALPS 1.0'
PACKAGE_TARNAME='alps'
PACKAGE_URL=''
PACKAGE_VERSION='1.0'
PATH_SEPARATOR=':'
SET_MAKE=''
SHELL='/bin/sh'
STRIP=''
VERSION='1.0'
ac_ct_CC='gcc'
ac_ct_FC='mpifort'
am__EXEEXT_FALSE=''
am__EXEEXT_TRUE='#'
am__fastdepCC_FALSE=''
am__fastdepCC_TRUE='#'
am__include='include'
am__isrc=''
am__leading_dot='.'
am__nodep='_no'
am__quote=''
am__tar='$${TAR-tar} chof - "$$tardir"'
am__untar='$${TAR-tar} xf -'
bindir='${exec_prefix}/bin'
build_alias=''
datadir='${datarootdir}'
datarootdir='${prefix}/share'
docdir='${datarootdir}/doc/${PACKAGE_TARNAME}'
dvidir='${docdir}'
exec_prefix='${prefix}'
host_alias=''
htmldir='${docdir}'
includedir='${prefix}/include'
infodir='${datarootdir}/info'
install_sh='${SHELL} /Users/joejiang/documents/ALPS/ALPS-master/install-sh'
libdir='${exec_prefix}/lib'
libexecdir='${exec_prefix}/libexec'
localedir='${datarootdir}/locale'
localstatedir='${prefix}/var'
mandir='${datarootdir}/man'
mkdir_p='$(MKDIR_P)'
oldincludedir='/usr/include'
pdfdir='${docdir}'
prefix='/usr/local'
program_transform_name='s,x,x,'
psdir='${docdir}'
sbindir='${exec_prefix}/sbin'
sharedstatedir='${prefix}/com'
sysconfdir='${prefix}/etc'
target_alias=''

## ----------- ##
## confdefs.h. ##
## ----------- ##

/* confdefs.h */
#define PACKAGE_NAME "ALPS"
#define PACKAGE_TARNAME "alps"
#define PACKAGE_VERSION "1.0"
#define PACKAGE_STRING "ALPS 1.0"
#define PACKAGE_BUGREPORT "d.verscharen@ucl.ac.uk"
#define PACKAGE_URL ""
#define PACKAGE "alps"
#define VERSION "1.0"

configure: exit 0

**Comment by danielver02** [comment]
Hi Joe,

are you sure that the compilation didn’t work? The compiler didn’t seem to report an error when linking. Maybe you have the executable already. It would be ./src/ALPS. If it’s there, check whether you can mpirun it as described in the documentation. 

The configure script isn’t happy with your Lapack installation. I’m not sure how to fix it though. 

Best wishes,
-Daniel

**Reply by joejiang2023** [reply]
Hi Daniel,

Thanks! I can run the mpirun -np 4 ./src/ALPS ./tests/test_ICW.in
But it still reports an error in the distribution generation part, which is related to the "dgesv" issue.
I need to find out how to link my LAPACK libraries to gfortran though.
-Joe

**Comment by joejiang2023** [comment, marked as answer]
Hi all,

For anyone run into a similar problem, I installed openBLAS and export these after installation:

  export LDFLAGS="-L/usr/local/opt/openblas/lib"
  export CPPFLAGS="-I/usr/local/opt/openblas/include"
  export PKG_CONFIG_PATH="/usr/local/opt/openblas/lib/pkgconfig"

then I run ./configure in the ALPS directory and successfully link the dgesv routine to the gfortran.

-Joe
